### PR TITLE
[BACKUP] Add Cross Subscription Restore support for SQL/HANA and V1 Vaults

### PR DESCRIPTION
**Pull request by akshayneema**
**Related command**

-  `az backup vault create`
- `az backup recoveryconfig show`

**Description**

- `az backup vault create`: Add parameter `cross-subscription-restore-state` which can be used to set the CSR state of the vault at the time of creation as well as updation
-  `az backup recoveryconfig show`: Add parameter `--target-subscription-id` which can be used to provide the target subscription as the input while triggering cross subscription restore for SQL or HANA workloads

**Testing Guide**

`az backup vault create -g {rg_name} -n {vault_name} -l {location} --cross-subscription-restore-state Disable`

 `az backup recoveryconfig show --restore-mode alternateworkloadrestore --backup-management-type azureworkload -r {rp} --target-container-name {target_container} --target-item-name {target_item} --target-resource-group {target_rg} --target-server-name {target_server} --target-server-type SQLInstance --target-subscription-id {target_subscription} --target-vault-name {target_vault} --workload-type  SQLDataBase --ids {source_item_id}`

**History Notes**

[BACKUP] `az backup vault create`: Add parameter `--cross-subscription-restore-state` to set the CSR state of the vault at the time of creation as well as updating
[BACKUP] `az backup recoveryconfig show`: Add parameter `--target-subscription-id` to provide the target subscription as the input while triggering cross subscription restore for SQL or HANA workloads

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

File: src/azure-cli/azure/cli/command_modules/backup/_help.py [hunk 1]
```diff
@@ -473,9 +473,9 @@
 
 helps['backup vault create'] = """
 type: command
-short-summary: Create a new Recovery Services vault.
+short-summary: Create a new Recovery Services vault or Update an existing one.
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/backup/_help.py:476]:
```suggestion
short-summary: Create a new Recovery Services vault or update an existing one.
```

File: src/azure-cli/azure/cli/command_modules/backup/_help.py [hunk 2]
```diff
@@ -473,9 +473,9 @@
...
 examples:
-  - name: Create a new Recovery Services vault. (autogenerated)
+  - name: Create a new Recovery Services vault or Update an existing one. (autogenerated)
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/backup/_help.py:478]:
```suggestion
  - name: Create a new Recovery Services vault or update an existing one. (autogenerated)
```

File: src/azure-cli/azure/cli/command_modules/backup/_params.py [hunk 3]
```diff
@@ -414,6 +415,7 @@ def load_arguments(self, _):
         c.argument('backup_management_type', backup_management_type)
         c.argument('target_resource_group', options_list=['--target-resource-group'], help="""Specify the resource group of target item for Cross Region Restore. Default value will be same as --resource-group if not specified.""")
         c.argument('target_vault_name', options_list=['--target-vault-name'], help="""Specify the vault name of target item for Cross Region Restore. Default value will be same as --vault-name if not specified.""")
+        c.argument('target_subscription_id', options_list=['--target-subscription-id'], help="""Specify the subscription of the target item for Cross Subscription Restore. Defaulted to source subscription if not specified.""")
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/backup/_params.py:418]: We do not need to explicitly specify the default option name
```suggestion
        c.argument('target_subscription_id', help="""Specify the subscription of the target item for Cross Subscription Restore. Defaulted to source subscription if not specified.""")
```

File: src/azure-cli/azure/cli/command_modules/backup/custom_help.py [hunk 4]
```diff
@@ -521,6 +521,11 @@ def get_resource_group_from_id(arm_id):
     return m.group(0)
 
 
+def get_subscription_from_id(arm_id):
+    m = re.search('(?<=/subscriptions/)[^/]+', arm_id)
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/backup/custom_help.py:525]: Please add comments or examples for the regular expressions

Comment (akshayneema) [reply src/azure-cli/azure/cli/command_modules/backup/custom_help.py:528]: Added comments to explain the regex

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**